### PR TITLE
Move Lookup namespace from grain_size to material_model/utilities

### PR DESCRIPTION
**Pull request by pmbremner**
This change moves the entire Lookup namespace, which contains the class
MaterialLookup from grain_size.cc and grain_size.h over to
material_model/utilities.cc and utilities.h. The functions contained in
this namespace are used in the grain_size and steinberger material
models, and potentially more in the future. The changes to the
steinberger material model are part of separate commit.

### CODE REVIEW

File: include/aspect/material_model/utilities.h [hunk 1]
```diff
@@ -41,6 +41,161 @@ namespace aspect
      */
     namespace MaterialUtilities
     {
+      namespace Lookup
+      {
+
+
+
```
Comment (gassmoeller) [inline include/aspect/material_model/utilities.h]: Please remove these empty lines

File: include/aspect/material_model/utilities.h [hunk 2]
```diff
@@ -41,6 +41,161 @@ namespace aspect
...
+         * and functions to access this data, but it does not contain the functions
+         * to read this data, which has to be implemented in a derived class.
+         */
+        class MaterialLookup
+        {
+          public:
+
+            double
+            specific_heat(double temperature,
+                          double pressure) const;
```
Comment (gassmoeller) [inline include/aspect/material_model/utilities.h]: not your fault, but change the two `double` arguments to `const double`. Same in the function below.

File: include/aspect/material_model/utilities.h [hunk 3]
```diff
@@ -171,6 +326,8 @@ namespace aspect
       compute_drucker_prager_yielding (const DruckerPragerInputs &in,
                                        DruckerPragerOutputs &out);
 
+
```
Comment (gassmoeller) [inline include/aspect/material_model/utilities.h]: Please remove these two new empty lines.

File: source/material_model/utilities.cc [hunk 4]
```diff
@@ -35,6 +35,545 @@ namespace aspect
   {
     namespace MaterialUtilities
     {
+      namespace Lookup
+      {
+
+
+
```
Comment (gassmoeller) [inline source/material_model/utilities.cc]: Please remove the empty lines after the opening curly brace

File: source/material_model/utilities.cc [hunk 5]
```diff
@@ -35,6 +35,545 @@ namespace aspect
...
+        double
+        MaterialLookup::specific_heat(double temperature,
+                                      double pressure) const
```
Comment (gassmoeller) [inline source/material_model/utilities.cc]: same comment, please adjust to `const double` here and in the function below

File: source/material_model/utilities.cc [hunk 6]
```diff
@@ -213,6 +752,9 @@ namespace aspect
 
         return;
       }
+
+
+
```
Comment (gassmoeller) [inline source/material_model/utilities.cc]: Please remove these empty lines

**Comment by pmbremner** [comment]
Addressed the comments

**Comment by gassmoeller** [comment]
You need to fix `tests/latent_heat_enthalpy.cc`, but otherwise looks good.

**Comment by pmbremner** [comment]
Addressed comments

**Comment by pmbremner** [comment]
Fixed conflict with latent_heat_enthalpy.cc again

**Comment by gassmoeller** [comment]
Great, thanks for the cleanup! Ready to merge.

**Pull request merged**